### DISCUSSION TITLE
Problem with Package-X in FeynHelpers

### DISCUSSION THREAD

**Discussion by moatms** [FeynCalc/feyncalc#172] [Q&A] [answered]
Hello,

I am sorry if that is not the right place to ask my question, but I couldn't find a more appropriate place. I am trying to calculate 1-loop amplitudes using FeynCalc, FeynHelpers and Package-X, and then I am trying to do a series expansion on the amplitude. To calculate the loop, I used:
AmpFCv2 = AmpFC // TID[#, k, ToPaVe -> True]&
AmpFCv3 = 
 PaXEvaluate[AmpFCv2, k, PaXD0Expand -> True, 
  PaXAnalytic -> True, PaXC0Expand -> True)]

And in the resulting amplitude, I get multiple terms with a "PaXContinuedDiLog", for example: PaXContinuedDiLog[
    {(s/x^2 + 2MH^2)/MH^2, -(s/x^2) - MH^2}, 
    {(-(s/x^2) + (k*s)/x^2 + 2*MH^2)/(2*MH^2), 
     s/x^2 - (k*s)/x^2}]
At this point, I would like to do a series expansion around x, but if I evaluate SeriesCoefficient[T1,{x,0,0}], where T1 is my PaXContinuedDiLog term just above, then Mathematica doesn't seem to understand this PaXContinuedDiLog function and simply returns:
SeriesCoefficient[PaXContinuedDiLog[
    {(s/x^2 + 2*MH^2)/MH^2, -(s/x^2) - MH^2}, 
    {(-(s/x^2) + (k*s)/x^2 + 2*MH^2)/(2*MH^2), 
     s/x^2 - (k*s)/x^2}],{x,0,0}]. 
In the same way, if I try to get the derivative of this PaxContinuedDiLog term, Mathematica just returns a general formula with many PaXContinuedDiLog and doesn't seem to understand how to evaluate the function.
Is there any way to evaluate PaXContinuedDiLog  more explicitly to then be able to use SeriesCoefficient on it?

Many thanks!

**Comment by vsht** [comment, marked as answer]
Hi,

I've just pushed a big update to the dev version that should (among other things) pass PaXContinuedDiLog (which is just a placeholder for
X`ContinuedDiLog) to Package-X when derivatives are requested.

https://feyncalc.github.io/FeynHelpersBookDev/PaXContinuedDiLog.html

However, I'm not entirely sure whether the expansion you want to do is implemented in Package-X itself.
Here one just gets back the original expression without a `SeriesData` head
```
<<X`

Series[ContinuedDiLog[{(s/x^2 + 2 MH^2)/MH^2, -(s/x^2) - 
    MH^2}, {(-(s/x^2) + (ks)/x^2 + 2 MH^2)/(2 MH^2), 
   s/x^2 - (k*s)/x^2}], {x, 0, 2}]
(*ContinuedDiLog[{(2*MH^2 + ks/x^2 - s/x^2)/(2*MH^2), s/x^2 - (k*s)/x^2}, {(2*MH^2 + s/x^2)/MH^2, -MH^2 - s/x^2}]*)
```
while in the case of the opposite limit the expansion clearly works
```
Series[ContinuedDiLog[{(s/x^2 + 2 MH^2)/MH^2, -(s/x^2) - 
    MH^2}, {(-(s/x^2) + (ks)/x^2 + 2 MH^2)/(2 MH^2), 
   s/x^2 - (k*s)/x^2}], {x, Infinity, 2}]
(*SeriesData[x, Infinity, {-1/12*Pi^2, 0, -((ks*Log[2])/MH^2)}, 0, 3, 1]*)
```
Perhaps it would be good to get in touch with Hiren to be 100%-sure regarding the reliability of the expansion.

**Reply by vsht** [reply]
BTW, he is also on GitHub as @MrHirenP :)

**Reply by moatms** [reply]
Ok, thanks a lot for your help!
